src/definition-finder.ts

```diff
@@ -0,0 +1,49 @@
+'use babel';
+
+import MouseMoveListener from './mouse-move-listener';
+import {SourceRange} from 'polymer-analyzer/lib/model/model'
+
+class DefinitionFinder extends MouseMoveListener {
+  sourcePosition: SourceRange;
+  removeMouseMoveListener: Function;
+
+  beforeUpdate() {
+    if (this.removeMouseMoveListener) {
+      this.removeMouseMoveListener();
+    }
+  };
+
+  obtainMouseMoveListener(textEditorElement: any) {
+    this.attachMouseDownListener(textEditorElement);
+    return async(event: MouseEvent) => {
+      const screenPosition = textEditorElement.component.screenPositionForMouseEvent(event);
+      const point = this.textEditor.bufferPositionForScreenPosition(screenPosition);
+      const relativePath: string =
+        atom.project.relativizePath(this.textEditor.getPath())[1];
+
+      try {
+        this.sourcePosition = await this.editorService.getDefinitionForFeatureAtPosition(relativePath, {line: point.row, column: point.column});
+      } catch (e) {
+        return;
+      }
+    };
+  };
+
+  private attachMouseDownListener(textEditorElement: any) {
+    console.log(textEditorElement);
+    const mouseDownListener = this.mouseDownListener.bind(this);
+    textEditorElement.addEventListener('mousedown', mouseDownListener);
+    this.removeMouseMoveListener = () => textEditorElement.removeEventListener('mousedown', mouseDownListener);
+  };
+
+  private async mouseDownListener(event: MouseEvent) {
+    if (this.sourcePosition && event.ctrlKey) {
```

Comment:
This is racy and the most likely source of the problem. This assumes that a different event handler has fired, that it has completed, and that its results are still relevant.

If you're doing it this way so that you'll have access to the textEditorElement and so can get the filename, then this should be a closure inside `attachMouseDownListener`. Then there's not this spooky asynchronous action at a distance.